enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextTransactionEventUpdater.java

```diff
@@ -87,77 +86,32 @@ public Object beforeCommit( TransactionData data ) throws Exception
     @Override
     public void afterCommit( TransactionData data, Object state )
     {
-        //update node indices
         try
         {
+            List<BinaryLatch> completions = new ArrayList<>();
```

Comment:
what do we gaining by generating lots independent small updates? why not batch them to at least one per index?

Reply:
The applier batches as it consumes from the queue, but you want to put the loop into the applier? We can do that.

Reply:
yeah basically generate only one update that ges to applier queue at least per index per commit